src/Services/InventoryUpdateService.php

```diff
@@ -37,319 +54,524 @@ class InventoryUpdateService
   const PAGES = 'pages';
   const ERROR_MESSAGE = 'errorMessage';
 
-  /**
-   * Validate a request for inventory update
-   * @param RequestDTO $inventoryRequestDTO
-   * @param LoggerContract $loggerContract
-   * @return bool
-   */
-  private function validateInventoryRequestData($inventoryRequestDTO, $loggerContract): bool
-  {
-    if (!isset($inventoryRequestDTO)) {
-      return false;
-    }
+  const VARIATIONS_PER_PAGE = 200;
 
-    $issues = [];
+  /** @var InventoryStatusService */
+  private $statusService;
 
-    $supplierId = $inventoryRequestDTO->getSupplierId();
+  /** @var InventoryService */
+  private $inventoryService;
 
-    if (!isset($supplierId) || $supplierId <= 0) {
-      $issues[] = "Supplier ID is missing or invalid";
-    }
-
-    $partNum = $inventoryRequestDTO->getSupplierPartNumber();
+  /** @var InventoryMapper */
+  private $inventoryMapper;
 
-    if (!isset($partNum) || empty($partNum)) {
-      $issues[] = "Supplier Part number is missing";
-    }
+  /** @var LoggerContract */
+  private $logger;
 
-    $onHand = $inventoryRequestDTO->getQuantityOnHand();
+  /** @var AbstractConfigHelper */
+  private $configHelper;
 
-    if (isset($onHand)) {
-      if ($onHand  < -1) {
-        $issues[] = "Quantity on Hand is less than negative one";
-      }
-    } else {
-      $issues[] = "Quantity On Hand is missing";
-    }
-
-    if (!isset($issues) || empty($issues)) {
-      return true;
-    }
-
-    // TODO: replace issues with translated messages?
-    $loggerContract
-      ->error(
-        TranslationHelper::getLoggerKey(self::LOG_KEY_INVALID_INVENTORY_DTO),
-        [
-          'additionalInfo' => [
-            'message' => 'inventory request data is invalid',
-            'issues' => json_encode($issues),
-            'data' => $inventoryRequestDTO->toArray(),
-          ],
-          'method' => __METHOD__
-        ]
-      );
+  /** @var LogSenderService */
+  private $logSenderService;
 
-    return false;
+  public function __construct(
+    InventoryService $inventoryService,
+    InventoryMapper $inventoryMapper,
+    InventoryStatusService $statusService,
+    AbstractConfigHelper $configHelper,
+    LoggerContract $logger,
+    LogSenderService $logSenderService
+  ) {
+    $this->inventoryService = $inventoryService;
+    $this->inventoryMapper = $inventoryMapper;
+    $this->statusService = $statusService;
+    $this->configHelper = $configHelper;
+    $this->logger = $logger;
+    $this->logSenderService = $logSenderService;
   }
 
   /**
-   * @param bool $fullInventory
+   * Send inventory updates from Plentymarkets to Wayfair.
+   * The optional start parameter filters the update set down to items changed since the provided timestamp
```

Comment:
Should there actually be a start parameter or is this no longer needed?

Reply:
ah yeah this comment is left over - the start time gets calculated dynamically and should no longer be provided by the caller.